sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultBackupClientImpl.java

```diff
@@ -199,6 +275,145 @@ public Mono<FullBackupResponse> fullBackupWithResponseAsync(
+     * @throws KeyVaultErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return full backup operation on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<FullBackupOperation> fullBackupAsync(
+            String vaultBaseUrl, SASTokenParameter azureStorageBlobContainerUri) {
+        return fullBackupWithResponseAsync(vaultBaseUrl, azureStorageBlobContainerUri)
+                .flatMap(res -> Mono.justOrEmpty(res.getValue()));
+    }
+
+    /**
+     * Creates a full backup using a user-provided SAS token to an Azure blob storage container. This operation is
+     * supported only by the Managed HSM service.
+     *
+     * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+     * @param azureStorageBlobContainerUri Azure blob shared access signature token pointing to a valid Azure blob
+     *     container where full backup needs to be stored. This token needs to be valid for at least next 24 hours from
+     *     the time of making this call.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws KeyVaultErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return full backup operation on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<FullBackupOperation> fullBackupAsync(
+            String vaultBaseUrl, SASTokenParameter azureStorageBlobContainerUri, Context context) {
+        return fullBackupWithResponseAsync(vaultBaseUrl, azureStorageBlobContainerUri, context)
+                .flatMap(res -> Mono.justOrEmpty(res.getValue()));
+    }
+
+    /**
+     * Creates a full backup using a user-provided SAS token to an Azure blob storage container. This operation is
+     * supported only by the Managed HSM service.
+     *
+     * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+     * @param azureStorageBlobContainerUri Azure blob shared access signature token pointing to a valid Azure blob
+     *     container where full backup needs to be stored. This token needs to be valid for at least next 24 hours from
+     *     the time of making this call.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws KeyVaultErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return full backup operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public FullBackupResponse fullBackupSyncWithResponse(
```

Comment:
nit; Why do async methods have the `Async` suffix after `WithResponse` but sync methods use `Sync` before it? This is probably something on Autorest's side and, since it's not a public class, won't matter to customers.

Reply:
Just autorest logic, @srnagar